app/assets/javascripts/models/page/project_page.js.coffee

```diff
@@ -139,24 +134,25 @@ ns.ProjectPage = do (ko,
         loadedCollectionNum = 0
         for projectCollection in @project.collections()
           collection = projectCollection.collection
-          granules = collection.cmrGranulesModel
-          if granules.isLoaded()
-            loadedCollectionNum += 1
-            _size = 0
-            _size += parseFloat(granule.granule_size ? 0) for granule in granules.results()
-            totalSize = _size / granules.results().length * granules.hits()
-            totalSize = 0 if isNaN(totalSize)
-            projectGranules += granules.hits()
-            collection.granule_hits(granules.hits())
-            projectSize += totalSize
-            if isNaN(totalSize) || (totalSize == 0 && granules.hits() > 0)
-              collection.total_size('Not Provided')
-              collection.unit('')
-              @sizeProvided(false)
-            else
-              collection.total_size(@_convertSize(totalSize)['size'])
-              collection.unit(@_convertSize(totalSize)['unit'])
-              @sizeProvided(true)
+          if collection.granuleDatasource()
+            granules = collection.granuleDatasource().data()
+            if granules.isLoaded()
+              loadedCollectionNum += 1
+              _size = 0
+              _size += parseFloat(granule.granule_size ? 0) for granule in granules.results()
+              totalSize = _size / granules.results().length * granules.hits()
+              totalSize = 0 if isNaN(totalSize)
+              projectGranules += granules.hits()
+              collection.granule_hits(granules.hits())
+              projectSize += totalSize
+              if isNaN(totalSize) || (totalSize == 0 && granules.hits() > 0)
```

Comment:
isNaN(totalSize) will never be true here, as totalSize is set to 0 in the case that it is NaN on previous line 144.  Since that part of the conditional will never be true, this could probably be simplified to:

```
if (totalSize == 0 && granules.hits() > 0)
```